src/main/scala/chisel3/util/BlackBoxUtils.scala

```diff
@@ -4,12 +4,15 @@ package chisel3.util
 
 import chisel3._
 import chisel3.experimental.{ChiselAnnotation, RunFirrtlTransform}
-import firrtl.transforms.{BlackBoxResourceAnno, BlackBoxInlineAnno, BlackBoxSourceHelper}
+import firrtl.transforms.{BlackBoxPathAnno, BlackBoxResourceAnno, BlackBoxInlineAnno, BlackBoxSourceHelper}
 
 trait HasBlackBoxResource extends BlackBox {
   self: BlackBox =>
 
-  def setResource(blackBoxResource: String): Unit = {
+  @deprecated("Use addResource instead", "3.2")
+  def setResource(blackBoxResource: String): Unit = addResource(blackBoxResource)
+
+  def addResource(blackBoxResource: String): Unit = {
```

Comment:
I think it would be nice to add some scaladoc here, in particular, providing an example
of what a resource path looks like.